### ISSUE TITLE
Transform functions should bypass functionConfigs

### ISSUE THREAD

**Issue by prachirp** [kptdev/kpt#1122]
We don't want functions to manipulate the function config if it's in the same directory as KRM resources.

We need a way to bypass this.

Pain point example: https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/87

**Comment by mortent** [comment]
@prachirp So this the idea here that resources that have the `config.k8s.io/function` annotation should not be mutated by a function pipeline? There are scenarios where we want to put the config annotation on normal resources, so I'm not sure if that would work. This seems very similar to GoogleContainerTools/kpt#630. 
Another option might be that for resource that are only present to declaratively define a KRM function should not be of the ConfigMap type, but instead we could use a client-side CR where the Kind would be `Function`.

**Comment by prachirp** [comment]
@mortent Yes, that's a good summary. And I agree #630 is similar but not the same. I'm not sure what the solution should be here. It seems just defining a new annotation is a much simpler fix than using a new `Function` kind for function configs.

From Frank: #630 is a different issue, it's about the inventory object being processed. Function configs are different. Let's file a separate bug for this.

**Comment by aravindtga** [comment]
This issue is partially addressed:

  - **`fn eval`**: Fixed. PR #1767 (2021) originally excluded fn-config from items, PR #2894 (2022) reverted it by including all meta resources by default, then PR #4639 (2026) re-implemented the exclusion with a different approach.

  - **`fn render`**: Still affected. Function configs specified via `configPath` are included in the mutator input and can be mutated by pipeline functions. **Workaround**: use `exclude` selectors in the Kptfile.

 Was PR #2894's removal of fn-config exclusion intentional for `fn eval`, or an unintended side-effect? And should `fn render` also exclude `configPath` resources from the pipeline items to match the documented behaviour?

cc/ @liamfallon @efiacor @CsatariGergely @mozesl-nokia

**Comment by CsatariGergely** [comment]
Functions should not act on their own configurations. 
We should document this somewhere. We should fix `fn render` in the scope of this issue.
